src/api/goods.ts

```
@@ -3,14 +3,47 @@ import { Good } from '../types/Good';
 // eslint-disable-next-line
 const API_URL = `https://mate-academy.github.io/react_dynamic-list-of-goods/goods.json`;
 
-export function getAll(): Promise<Good[]> {
-  return fetch(API_URL).then(response => response.json());
+// Função para simular atraso, útil para exibir Loader
+const wait = (ms: number) => new Promise(resolve => setTimeout(resolve, ms));
+
+// Retorna a lista completa de produtos
+export async function getAllGoods(): Promise<Good[]> {
+  try {
+    const res = await fetch(API_URL);
+    const data = await res.json();
+
+    await wait(500); // delay artificial para mostrar Loader
+
+    return Array.isArray(data) ? data : [];
+  } catch {
+    return [];
+  }
 }
 
-export const get5First = () => {
-  return getAll().then(goods => goods); // sort and get the first 5
-};
+// Retorna os 5 primeiros produtos ordenados por nome
+export async function getFirstFiveSortedByName(): Promise<Good[]> {
+  try {
+    const goods = await getAllGoods();
 
-export const getRedGoods = () => {
-  return getAll().then(goods => goods); // get only red
-};
+    return [...goods]
+      .sort((a, b) =>
+        (a.name ?? '').toString().localeCompare((b.name ?? '').toString()),
+      )
+      .slice(0, 5);
+  } catch {
+    return [];
+  }
```

Issue: getFirstFiveSortedByName wraps its logic in try/catch and returns [] on error. If getAllGoods throws on network failure (see recommendation above), this helper will swallow that error and return [], making the UI believe the request succeeded with an empty list. This violates the checklist item: "3. [API] - When you make a request to the server, don't forget to handle possible errors" . Allow errors to propagate to App.tsx (remove the outer try/catch or rethrow inside it).